tests/test_models/test_inference.py

```diff
@@ -92,434 +143,982 @@ def _test_forecast_out_sample_prefix(ts, model, transforms):
+
+    @pytest.mark.xfail(strict=True)
+    @pytest.mark.parametrize(
+        "model, transforms",
+        [
+            (CatBoostModelPerSegment(), [LagTransform(in_column="target", lags=[2, 3])]),
+            (LinearPerSegmentModel(), [LagTransform(in_column="target", lags=[2, 3])]),
+            (LinearMultiSegmentModel(), [LagTransform(in_column="target", lags=[2, 3])]),
+            (ElasticPerSegmentModel(), [LagTransform(in_column="target", lags=[2, 3])]),
+            (ElasticMultiSegmentModel(), [LagTransform(in_column="target", lags=[2, 3])]),
+        ],
+    )
+    def test_forecast_in_sample_full_no_target_failed(self, model, transforms, example_tsds):
+        _test_forecast_in_sample_full_no_target(example_tsds, model, transforms)
+
+    @pytest.mark.parametrize(
+        "model, transforms",
+        [
+            (MovingAverageModel(window=3), []),
+            (NaiveModel(lag=3), []),
+            (SeasonalMovingAverageModel(), []),
+            (DeadlineMovingAverageModel(window=1), []),
+        ],
+    )
+    def test_forecast_in_sample_full_no_target_failed_not_enough_context(self, model, transforms, example_tsds):
+        with pytest.raises(ValueError, match="Given context isn't big enough"):
+            _test_forecast_in_sample_full_no_target(example_tsds, model, transforms)
+
+    @pytest.mark.parametrize(
+        "model, transforms",
+        [
+            (BATSModel(use_trend=True), []),
+            (TBATSModel(use_trend=True), []),
+            (
+                DeepARModel(max_epochs=1, learning_rate=[0.01]),
+                [
+                    PytorchForecastingTransform(
+                        max_encoder_length=1,
+                        max_prediction_length=1,
+                        time_varying_known_reals=["time_idx"],
+                        time_varying_unknown_reals=["target"],
+                        target_normalizer=GroupNormalizer(groups=["segment"]),
+                    )
+                ],
+            ),
+            (
+                TFTModel(max_epochs=1, learning_rate=[0.01]),
+                [
+                    PytorchForecastingTransform(
+                        max_encoder_length=21,
+                        min_encoder_length=21,
+                        max_prediction_length=5,
+                        time_varying_known_reals=["time_idx"],
+                        time_varying_unknown_reals=["target"],
+                        static_categoricals=["segment"],
+                        target_normalizer=None,
+                    )
+                ],
+            ),
+        ],
+    )
+    def test_forecast_in_sample_full_no_target_not_implemented_in_sample(self, model, transforms, example_tsds):
+        with pytest.raises(NotImplementedError, match="It is not possible to make in-sample predictions"):
+            _test_forecast_in_sample_full_no_target(example_tsds, model, transforms)
+
+
+class TestForecastInSampleFull:
```

Comment:
If you've started to use Classes may be we should define helper functions like `_test_predict_out_sample` there. 
It seems we can logically separate them too.

Reply:
There are two helper functions like: `_test_prediction_in_sample_full` and `_test_prediction_in_sample_suffix` that accept `method_name` parameter and can work with both `forecast` and `predict`. Where do we want to put them?